ur_robot_driver/test/integration_test.py

```diff
@@ -87,6 +91,15 @@ def init_robot(self):
                 "Could not reach joint passthrough controller action. Make sure that the driver is actually running."
                 " Msg: {}".format(err))
 
+        self.cartesian_trajectory_client = actionlib.SimpleActionClient(
+            'follow_cartesian_trajectory', FollowCartesianTrajectoryAction)
+        try:
```

Comment:
I guess `wait_for_server` does not throw but [returns false on timeout](https://docs.ros.org/en/noetic/api/actionlib/html/simple__action__client_8py_source.html#l00067).
Maybe simply use:

```python
        if not self.cartesian_trajectory_client.wait_for_server(timeout):
            self.fail("Could not reach cartesian controller action. Make sure that the driver is actually running.")

```

Reply:
Thanks for spotting this. `ServiceProxy.wait_for_service` [throws an exception](https://github.com/ros/ros_comm/blob/noetic-devel/clients/rospy/src/rospy/impl/tcpros_service.py#L93), while  [`ActionClient.wait_for_server`](https://docs.ros.org/en/noetic/api/actionlib/html/action__client_8py_source.html#l00590) does not.

Reply:
Resolved in 65e55ac